### PR TITLE
Sol notifs

### PR DESCRIPTION
**Pull request by Sosol02**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/notifs/NotificationPublisher.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import android.content.Intent;
+import android.util.Log;
+import android.widget.Toast;
+
+public class NotificationPublisher extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d("NotificationPublisher", "onReceive");
+        Toast.makeText(context, "I am a toast owo", Toast.LENGTH_LONG).show();
```
Comment (hugolan) [inline app/src/main/java/com/github/onedirection/notifs/NotificationPublisher.java:16]: This was testing I suppose, try to remove it for next time

**Review by hugolan — APPROVED** [review]

File: app/src/androidTest/java/com/github/onedirection/notifs/NotificationsTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public void tearDown() {
+        Intents.release();
+    }
+
+    public void gotoCreator() {
+        onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
+        onView(withId(R.id.nav_create_event)).perform(ViewActions.click());
+    }
+
+    private void setTimePicker(int startId, ZonedDateTime time) {
```
Comment (Flechman) [inline app/src/androidTest/java/com/github/onedirection/notifs/NotificationsTest.java:74]: Check time null

File: app/src/androidTest/java/com/github/onedirection/notifs/NotificationsTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        onView(withId(startId)).perform(ViewActions.click());
+        onView(withClassName(equalTo(TimePicker.class.getName()))).perform(PickerActions.setTime(time.getHour(), time.getMinute()));
+        onView(withId(android.R.id.button1)).perform(ViewActions.click());
+    }
+
+    private void setDatePicker(int startId, ZonedDateTime date) {
```
Comment (Flechman) [inline app/src/androidTest/java/com/github/onedirection/notifs/NotificationsTest.java:80]: Check time null

File: app/src/main/java/com/github/onedirection/notifs/NotificationPublisher.java (hunk 1 above)
Comment (Sosol02) [reply app/src/main/java/com/github/onedirection/notifs/NotificationPublisher.java:16]: Oh right, will do

File: app/src/main/java/com/github/onedirection/notifs/Notifications.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        if (global == null) {
+            setupNotifChannel(context);
+            global = new Notifications();
+        }
+        return global;
+    }
+
+    private static final String CHANNEL_ID = "todo";
+
+    private static void setupNotifChannel(Context context) {
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/notifs/Notifications.java:52]: Check context

File: app/src/main/java/com/github/onedirection/notifs/Notifications.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            channel.setDescription(description);
+            NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    private final Set<Event> events = new HashSet<>();
+    private Optional<Event> currentlyScheduled = Optional.empty();
+
+    public void schedule(Context context, long whenMillis, int notifId, /* TODO fix that */ Parcelable payload, BroadcastReceiver handler) {
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/notifs/Notifications.java:67]: Check parameters

**Review by Flechman — APPROVED** [review]
Working, but for next time, check all your method parameters

**Pull request merged**